### PR TITLE
Remove unused lint exceptions

### PR DESCRIPTION
**Pull request by jablko**
- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## angular-material ([<kbd>unpkg</kbd>](https://unpkg.com/browse/angular-material@latest/))
was missing the following properties:
1. padStart
1. padEnd
1. trimLeft
1. trimRight
1. trimStart

<details>
<summary>as well as these 7 other properties...</summary>
<p>trimEnd, padStart, padEnd, trimLeft, trimRight, trimStart, trimEnd</p>
</details>

## babel-plugin-react-html-attrs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/babel-plugin-react-html-attrs@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'babel-plugin-react-html-attrs'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.
## bluebird ([<kbd>unpkg</kbd>](https://unpkg.com/browse/bluebird@latest/))
was missing the following properties:
1. TypeError
1. RangeError
1. RejectionError
1. fulfilled
1. rejected

<details>
<summary>as well as these 1 other properties...</summary>
<p>pending</p>
</details>

## c3 ([<kbd>unpkg</kbd>](https://unpkg.com/browse/c3@latest/))
was missing the following properties:
1. chart
## documentdb ([<kbd>unpkg</kbd>](https://unpkg.com/browse/documentdb@latest/))
was missing the following properties:
1. DocumentBase
1. Base
1. Constants
1. RetryOptions
## feathersjs__socket-commons ([<kbd>unpkg</kbd>](https://unpkg.com/browse/feathersjs__socket-commons@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'feathersjs__socket-commons'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
## figlet ([<kbd>unpkg</kbd>](https://unpkg.com/browse/figlet@latest/))
was missing the following properties:
1. parseFont
1. preloadFonts
1. figFonts
## google-cloud__datastore ([<kbd>unpkg</kbd>](https://unpkg.com/browse/google-cloud__datastore@latest/))
was missing the following properties:
1. double
1. isDouble
1. geoPoint
1. isGeoPoint
1. int

<details>
<summary>as well as these 3 other properties...</summary>
<p>isInt, isKey, v1</p>
</details>

## google-protobuf ([<kbd>unpkg</kbd>](https://unpkg.com/browse/google-protobuf@latest/))
was missing the following properties:
1. exportSymbol
1. inherits
1. object
1. typeOf
## google__maps ([<kbd>unpkg</kbd>](https://unpkg.com/browse/google__maps@latest/))
was missing the following properties:
1. cli
1. util
1. cli
1. util
## hapi ([<kbd>unpkg</kbd>](https://unpkg.com/browse/hapi@latest/))
was missing the following properties:
1. server
## http-server ([<kbd>unpkg</kbd>](https://unpkg.com/browse/http-server@latest/))
was missing the following properties:
1. HttpServer
1. HTTPServer
## i18n ([<kbd>unpkg</kbd>](https://unpkg.com/browse/i18n@latest/))
was missing the following properties:
1. addLocale
1. removeLocale
## ink-select-input ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ink-select-input@latest/))
was missing the following properties:
1. Item
1. Indicator
## inputmask ([<kbd>unpkg</kbd>](https://unpkg.com/browse/inputmask@latest/))
was missing the following properties:
1. escapeRegex
1. keyCode
1. dependencyLib
## koa-bunyan-logger ([<kbd>unpkg</kbd>](https://unpkg.com/browse/koa-bunyan-logger@latest/))
was missing the following properties:
1. bunyan
## lambda-tester ([<kbd>unpkg</kbd>](https://unpkg.com/browse/lambda-tester@latest/))
was missing the following properties:
1. checkForResourceLeak
1. noVersionCheck
1. isVersionCheck
## mithril ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mithril@latest/))
was missing the following properties:
1. m
1. vnode
1. PromisePolyfill
## nightwatch ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nightwatch@latest/))
was missing the following properties:
1. client
1. cli
1. runner
1. runTests
1. CliRunner

<details>
<summary>as well as these 2 other properties...</summary>
<p>initClient, Logger</p>
</details>

## node-xmpp-client ([<kbd>unpkg</kbd>](https://unpkg.com/browse/node-xmpp-client@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'node-xmpp-client'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. NS_CLIENT
1. SASL
## parsimmon ([<kbd>unpkg</kbd>](https://unpkg.com/browse/parsimmon@latest/))
was missing the following properties:
1. fantasy-land/empty
1. fantasy-land/of
1. Binary
## powerbi-visuals-tools ([<kbd>unpkg</kbd>](https://unpkg.com/browse/powerbi-visuals-tools@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'powerbi-visuals-tools'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.
1. createVisualPackage
1. loadVisualPackage
1. installPackages
## prismjs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/prismjs@latest/))
was missing the following properties:
1. manual
1. disableWorkerMessageHandler
1. matchGrammar
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. sequence
## react-relay ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-relay@latest/))
was missing the following properties:
1. ConnectionHandler
1. MutationTypes
1. RangeOperations
1. ConnectionHandler
1. MutationTypes

<details>
<summary>as well as these 1 other properties...</summary>
<p>RangeOperations</p>
</details>

## reactstrap ([<kbd>unpkg</kbd>](https://unpkg.com/browse/reactstrap@latest/))
was missing the following properties:
1. DropdownContext
1. CustomFileInput
1. PopperContent
1. PopperTargetHelper
1. Polyfill

<details>
<summary>as well as these 5 other properties...</summary>
<p>DropdownContext, CustomFileInput, PopperContent, PopperTargetHelper, Polyfill</p>
</details>

## relay-runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/relay-runtime@latest/))
was missing the following properties:
1. ReplaySubject
1. MutationTypes
1. RangeOperations
1. MutationHandlers
1. VIEWER_ID

<details>
<summary>as well as these 9 other properties...</summary>
<p>VIEWER_TYPE, PreloadableQueryRegistry, createPayloadFor3DField, RelayError, DEFAULT_HANDLE_KEY, isScalarAndEqual, recycleNodesInto, stableCopy, getFragmentIdentifier</p>
</details>

## selenium-webdriver ([<kbd>unpkg</kbd>](https://unpkg.com/browse/selenium-webdriver@latest/))
was missing the following properties:
1. ThenableWebDriver
## slate ([<kbd>unpkg</kbd>](https://unpkg.com/browse/slate@latest/))
was missing the following properties:
1. Change
1. Node
1. Operation
1. resetMemoization
1. TextUtils

<details>
<summary>as well as these 1 other properties...</summary>
<p>useMemoization</p>
</details>

## styled-system ([<kbd>unpkg</kbd>](https://unpkg.com/browse/styled-system@latest/))
was missing the following properties:
1. overflowY
1. overflowX
1. overflowY
1. overflowX
## swiper ([<kbd>unpkg</kbd>](https://unpkg.com/browse/swiper@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'swiper'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. extendDefaults
## urijs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/urijs@latest/))
was missing the following properties:
1. version
1. protocol_expression
1. idn_expression
1. punycode_expression
1. ip4_expression

<details>
<summary>as well as these 19 other properties...</summary>
<p>ip6_expression, find_uri_expression, findUri, defaultPorts, hostProtocols, invalid_hostname_characters, domAttributes, getDomAttribute, characters, decodePath, decodeUrnPath, recodePath, recodeUrnPath, buildQueryParameter, setQuery, hasQuery, ensureValidHostname, ensureValidPort, noConflict</p>
</details>

## webpack ([<kbd>unpkg</kbd>](https://unpkg.com/browse/webpack@latest/))
was missing the following properties:
1. WebpackOptionsDefaulter
1. WebpackOptionsApply
1. NodeEnvironmentPlugin
1. validate
1. validateSchema

<details>
<summary>as well as these 6 other properties...</summary>
<p>WebpackOptionsValidationError, dependencies, web, webworker, node, util</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6b57c39c17e9b8c34c6880c5a87443e027f4fc93
</p>

**Comment by elibarzilay** [comment]
Besides the conflicts, there's the test failure which might be a result of touching too many packages...  Would you mind splitting it to smaller chunks?  (I know that some people had issues with the smaller prs, but if you'll do it this week, then I'll be the one who will merge them.)

**Comment by jablko** [comment]
@elibarzilay Okay, I rebased it and the tests are passing again. The least work for me would be to leave it as is, but I can do chunks if need be.

File: types/node/base.d.ts [hunk 1]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/base.d.ts:15]: This comment should stay

File: types/node/test/util.ts [hunk 2]
```diff
@@ -46,8 +46,6 @@ import { readFile } from 'fs';
 
     util.formatWithOptions({ colors: true }, 'See object %O', { foo: 42 });
 
-    // util.callbackify
```
Comment (elibarzilay) [inline types/node/test/util.ts:49]: Keep this

File: types/node/ts3.6/base.d.ts [hunk 3]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/ts3.6/base.d.ts:15]: Keep?  (Might be copy+paste, but better to be conservative)

File: types/node/v10/base.d.ts [hunk 4]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/v10/base.d.ts:15]: Keep

File: types/node/v12/base.d.ts [hunk 5]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/v12/base.d.ts:15]: same

File: types/node/v12/ts3.6/base.d.ts [hunk 6]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/v12/ts3.6/base.d.ts:15]: same

File: types/node/v13/base.d.ts [hunk 7]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/v13/base.d.ts:15]: same

File: types/node/v13/test/util.ts [hunk 8]
```diff
@@ -46,8 +46,6 @@ import { readFile } from 'fs';
 
     util.formatWithOptions({ colors: true }, 'See object %O', { foo: 42 });
 
-    // util.callbackify
```
Comment (elibarzilay) [inline types/node/v13/test/util.ts:49]: same

File: types/node/v13/ts3.6/base.d.ts [hunk 9]
```diff
@@ -12,8 +12,6 @@
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.bigint" />
 
-// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
```
Comment (elibarzilay) [inline types/node/v13/ts3.6/base.d.ts:15]: same

File: types/rax/index.d.ts [hunk 10]
```diff
@@ -442,8 +441,6 @@ declare namespace Rax {
    */
   type RaxInstance = Component<any> | Element;
 
-  // Base component for plain JS classes
```
Comment (elibarzilay) [inline types/rax/index.d.ts:445]: Keep

File: types/react/index.d.ts [hunk 11]
```diff
@@ -175,8 +175,6 @@ declare namespace React {
         ref: LegacyRef<T>;
     }
 
-    // ReactHTML for ReactHTMLElement
```
Comment (elibarzilay) [inline types/react/index.d.ts:178]: Keep

File: types/react/index.d.ts [hunk 12]
```diff
@@ -430,8 +427,6 @@ declare namespace React {
 
     type ReactInstance = Component<any> | Element;
 
-    // Base component for plain JS classes
```
Comment (elibarzilay) [inline types/react/index.d.ts:433]: Keep

File: types/react/v15/index.d.ts [hunk 13]
```diff
@@ -113,8 +113,6 @@ declare namespace React {
         ref: Ref<T>;
     }
 
-    // ReactHTML for ReactHTMLElement
```
Comment (elibarzilay) [inline types/react/v15/index.d.ts:116]: Keep

File: types/react/v15/index.d.ts [hunk 14]
```diff
@@ -295,8 +292,6 @@ declare namespace React {
 
     type ReactInstance = Component<any> | Element;
 
-    // Base component for plain JS classes
```
Comment (elibarzilay) [inline types/react/v15/index.d.ts:298]: Keep

File: types/screeps/screeps-tests.ts [hunk 15]
```diff
@@ -465,8 +462,6 @@ function resources(o: GenericStore): ResourceConstant[] {
                 }
             });
 
-            // Avoid creeps in the room
```
Comment (elibarzilay) [inline types/screeps/screeps-tests.ts:468]: Keep

File: types/sparqljs/index.d.ts [hunk 16]
```diff
@@ -281,8 +281,6 @@ export type Expression =
     | Tuple
     | Term;
 
-// allow Expression circularly reference itself
```
Comment (elibarzilay) [inline types/sparqljs/index.d.ts:284]: Keep?

File: types/sparqljs/v2/index.d.ts [hunk 17]
```diff
@@ -243,8 +243,6 @@ export type Expression =
     | Tuple
     | Term;
 
-// allow Expression circularly reference itself
```
Comment (elibarzilay) [inline types/sparqljs/v2/index.d.ts:246]: Keep?

File: types/webpack/v3/webpack-tests.ts [hunk 18]
```diff
@@ -127,11 +127,11 @@ configuration = {
 configuration = {
     externals: [
         // Disable TSLint for allowing non-arrow functions
```
Comment (elibarzilay) [inline types/webpack/v3/webpack-tests.ts:129]: Redundant?

**Review by elibarzilay — CHANGES_REQUESTED** [review]

File: types/node/base.d.ts (hunk 1 above)
Comment (jablko) [reply types/node/base.d.ts:15]: Done :heavy_check_mark:

File: types/node/test/util.ts (hunk 2 above)
Comment (jablko) [reply types/node/test/util.ts:49]: Done :heavy_check_mark:

File: types/node/ts3.6/base.d.ts (hunk 3 above)
Comment (jablko) [reply types/node/ts3.6/base.d.ts:15]: Done :heavy_check_mark:

File: types/node/v10/base.d.ts (hunk 4 above)
Comment (jablko) [reply types/node/v10/base.d.ts:15]: Done :heavy_check_mark:

File: types/node/v12/base.d.ts (hunk 5 above)
Comment (jablko) [reply types/node/v12/base.d.ts:15]: Done :heavy_check_mark:

File: types/node/v12/ts3.6/base.d.ts (hunk 6 above)
Comment (jablko) [reply types/node/v12/ts3.6/base.d.ts:15]: Done :heavy_check_mark:

File: types/node/v13/base.d.ts (hunk 7 above)
Comment (jablko) [reply types/node/v13/base.d.ts:15]: Done :heavy_check_mark:

File: types/node/v13/test/util.ts (hunk 8 above)
Comment (jablko) [reply types/node/v13/test/util.ts:49]: Done :heavy_check_mark:

File: types/node/v13/ts3.6/base.d.ts (hunk 9 above)
Comment (jablko) [reply types/node/v13/ts3.6/base.d.ts:15]: Done :heavy_check_mark:

File: types/rax/index.d.ts (hunk 10 above)
Comment (jablko) [reply types/rax/index.d.ts:445]: Done :heavy_check_mark:

File: types/react/index.d.ts (hunk 11 above)
Comment (jablko) [reply types/react/index.d.ts:178]: Done :heavy_check_mark:

File: types/react/index.d.ts (hunk 12 above)
Comment (jablko) [reply types/react/index.d.ts:433]: Done :heavy_check_mark:

File: types/react/v15/index.d.ts (hunk 13 above)
Comment (jablko) [reply types/react/v15/index.d.ts:116]: Done :heavy_check_mark:

File: types/react/v15/index.d.ts (hunk 14 above)
Comment (jablko) [reply types/react/v15/index.d.ts:298]: Done :heavy_check_mark:

File: types/screeps/screeps-tests.ts (hunk 15 above)
Comment (jablko) [reply types/screeps/screeps-tests.ts:468]: Done :heavy_check_mark:

File: types/sparqljs/index.d.ts (hunk 16 above)
Comment (jablko) [reply types/sparqljs/index.d.ts:284]: Done :heavy_check_mark:

File: types/sparqljs/v2/index.d.ts (hunk 17 above)
Comment (jablko) [reply types/sparqljs/v2/index.d.ts:246]: Done :heavy_check_mark:

**Pull request merged**